rst2pdf/extensions/inkscape_r2p.py

```diff
@@ -39,80 +36,118 @@ class InkscapeImage(VectorPdf):
 
     source_filecache = WeakKeyDictionary()
 
-    @classmethod
-    def available(self):
-        return True
-
-    def __init__(self, filename, width=None, height=None, kind='direct',
-                                 mask=None, lazy=True, srcinfo=None):
+    def __init__(
+        self,
+        filename,
+        width=None,
+        height=None,
+        kind='direct',
+        mask=None,
+        lazy=True,
+        srcinfo=None,
+    ):
         client, uri = srcinfo
         cache = self.source_filecache.setdefault(client, {})
         pdffname = cache.get(filename)
         if pdffname is None:
+            progname = shutil.which('inkscape')
             tmpf, pdffname = tempfile.mkstemp(suffix='.pdf')
             os.close(tmpf)
             client.to_unlink.append(pdffname)
             cache[filename] = pdffname
+
             # which version of inkscape?
             cmd = [progname, '--version']
             exitcode, out, err = InkscapeImage.run_cmd(cmd)
             if out.startswith('Inkscape 0.'):
                 # Inkscape 0.x uses -A
                 cmd = [progname, os.path.abspath(filename), '-A', pdffname]
             else:
-                # Inkscape 1.x uses --export-file
-                cmd = [progname, os.path.abspath(filename), '--export-file=%s' % pdffname]
+                # Inkscape 1.x uses --export-filename
+                cmd = [
+                    progname,
+                    os.path.abspath(filename),
+                    '--export-filename',
+                    pdffname,
+                ]
+
             try:
                 result = subprocess.call(cmd)
                 if result != 0:
                     log.error("Failed to run command: %s", ' '.join(cmd))
             except OSError:
                 log.error("Failed to run command: %s", ' '.join(cmd))
                 raise
+
             self.load_xobj((client, pdffname))
 
         pdfuri = uri.replace(filename, pdffname)
         pdfsrc = client, pdfuri
-        VectorPdf.__init__(self, pdfuri, width, height, kind, mask, lazy, pdfsrc)
+        VectorPdf.__init__(
+            self, pdfuri, width, height, kind, mask, lazy, pdfsrc
+        )
+
+    @classmethod
+    def available(self):
```

Comment:
Which code calls this method?

Reply:
I've no idea. I simply moved this. We probably want to run things through coverage.py at some point since I'm sure there's plenty of dead code here.

Reply:
Ah - I missed that it was at the top of the file…